Talos k8s with AWS CCM support

erikvveen opened the pull request.

PhilipSchmid commented on 00-locals.tf:
```diff
@@ -25,16 +25,23 @@ locals {
         extraArgs = {
           allocate-node-cidrs = var.allocate_node_cidrs
+          cloud-provider = "external"
+          leader-elect = true
```
```suggestion
```

This should already be the default, according to https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/.

PhilipSchmid commented on 00-locals.tf:
```diff
@@ -25,16 +25,23 @@ locals {
-          module.elb_k8s_elb.elb_dns_name
-        ]
+        extraArgs = {
+          cloud-provider = "external"
```
```suggestion
```

kube-apiserver doesn't have a `cloud-provider` extraArg, according to https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/.

PhilipSchmid commented on 00-locals.tf:
```diff
@@ -25,16 +25,23 @@ locals {
+        },
+        certSANs = [ 
+          module.elb_k8s_elb.elb_dns_name,
+          ]
```
Please fix the formatting

PhilipSchmid commented on 00-locals.tf:
```diff
@@ -59,17 +66,19 @@ locals {
   }
 
   # Used to configure Cilium Kube-Proxy replacement
+  
```
```suggestion
```

PhilipSchmid commented on 00-locals.tf:
```diff
@@ -85,13 +94,16 @@ locals {
+  
+  
+ 
+    config_patches_common = [
```
Restore indentation

PhilipSchmid commented on 00-variables.tf:
Remove this change, as it's already covered in your other PR: https://github.com/isovalent/terraform-aws-talos/pull/171

PhilipSchmid commented on 03-talos.tf:
```diff
@@ -10,6 +10,8 @@ module "talos_control_plane_nodes" {
   subnet_id                   = element(data.aws_subnets.public.ids, count.index)
   associate_public_ip_address = true
   tags                        = merge(var.tags, local.cluster_required_tags)
+  metadata_options            = var.metadata_options
```
Remove this change, as it's already covered in your other PR: https://github.com/isovalent/terraform-aws-talos/pull/171

PhilipSchmid commented on 03-talos.tf:
```diff
@@ -32,7 +34,9 @@ module "talos_worker_group" {
   associate_public_ip_address = true
   tags                        = merge(each.value.tags, var.tags, local.cluster_required_tags)
-
+  metadata_options            = var.metadata_options
```
Remove this change, as it's already covered in your other PR: https://github.com/isovalent/terraform-aws-talos/pull/171

PhilipSchmid commented on 03-talos.tf:
```diff
@@ -45,18 +49,32 @@ module "talos_worker_group" {
 resource "talos_machine_secrets" "this" {}
 
 data "talos_machine_configuration" "controlplane" {
+  count = var.controlplane_count
```
Could we iterate via the same method (`for_each`) as for the `worker_group` talos_machine_configuration?

PhilipSchmid commented on 03-talos.tf:
```diff
@@ -68,26 +86,39 @@ data "talos_machine_configuration" "worker_group" {
 resource "talos_machine_configuration_apply" "controlplane" {
   count = var.controlplane_count
-
+  
```
Remove spaces from all empty lines

PhilipSchmid commented on 03-talos.tf:
```diff
@@ -106,6 +137,7 @@ data "talos_client_configuration" "this" {
   cluster_name         = var.cluster_name
   client_configuration = talos_machine_secrets.this.client_configuration
   endpoints            = module.talos_control_plane_nodes.*.public_ip
+  nodes                = module.talos_control_plane_nodes.*.private_ip
```
Where does this come from? I don't see it in https://registry.terraform.io/providers/siderolabs/talos/0.7.0/docs/resources/machine_bootstrap. Same for `endpoints` (which was already there).

PhilipSchmid commented on 00-locals.tf:
```diff
@@ -25,16 +25,23 @@ locals {
-        ]
+        extraArgs = {
+          cloud-provider = "external"
+          enable-admission-plugins = "MutatingAdmissionWebhook,ValidatingAdmissionWebhook, ServiceAccount"
```
```suggestion
          enable-admission-plugins = "MutatingAdmissionWebhook,ValidatingAdmissionWebhook,ServiceAccount"
```

& please make that configurable.

PhilipSchmid reviewed (changes requested):
Thanks for adding this! I commented on a few things to address. Once done, please also run `terraform fmt`, `make docs`, and squash all commits into a single one with a meaningful message.

PhilipSchmid commented on 00-locals.tf:
```diff
@@ -25,16 +25,21 @@ locals {
 
   common_config_patch = {
     cluster = {
+
```
```suggestion
```

PhilipSchmid commented on 00-locals.tf:
```diff
@@ -59,17 +64,18 @@ locals {
       },
       certSANs = [
-        module.elb_k8s_elb.elb_dns_name
+        module.elb_k8s_elb.elb_dns_name,
```
```suggestion
        module.elb_k8s_elb.elb_dns_name
```

PhilipSchmid commented on 00-locals.tf:
```diff
@@ -25,16 +25,21 @@ locals {
         certSANs = [
-          module.elb_k8s_elb.elb_dns_name
-        ]
+          module.elb_k8s_elb.elb_dns_name,
```
```suggestion
          module.elb_k8s_elb.elb_dns_name
```

PhilipSchmid commented on 00-locals.tf:
```diff
@@ -59,17 +64,18 @@ locals {
   }
 
-  # Used to configure Cilium Kube-Proxy replacement
+  # Used to configure Cilium Kube-Proxy replacement  
```
```suggestion
  # Used to configure Cilium Kube-Proxy replacement
```

Remove trailing whitespaces

PhilipSchmid commented on 00-variables.tf:
```diff
@@ -9,6 +9,29 @@ variable "cluster_id" {
   type        = number
 }
 
+variable "iam_instance_profile_controller" {
```
```suggestion
variable "iam_instance_profile_control_plane" {
```

I'd like to be more specific about where this IAM profile is used - for the control plane notes.

PhilipSchmid commented on 00-variables.tf:
```diff
@@ -9,6 +9,29 @@ variable "cluster_id" {
 }
 
+variable "iam_instance_profile_controller" {
+  description = "IAM instance profile to attach to the controller instances."
```
```suggestion
  description = "IAM instance profile to attach to the control plane instances."
```

PhilipSchmid commented on 03-talos.tf:
```diff
@@ -10,6 +10,8 @@ module "talos_control_plane_nodes" {
   associate_public_ip_address = true
   tags                        = merge(var.tags, local.cluster_required_tags)
+  metadata_options            = var.metadata_options
+  iam_instance_profile        = var.iam_instance_profile_controller
```
```suggestion
  iam_instance_profile        = var.iam_instance_profile_control_plane
```

PhilipSchmid reviewed (commented):
Thanks for the updated version! I commented on a few nits (mostly with ready-to-apply suggestions). As soon as those are addressed, we can merge it.

PhilipSchmid reviewed (commented):
Have you tested this change without enabling your newly introduced variables? The example doesn't start anymore, because the CP nodes don't have the `node-role.kubernetes.io/control-plane=` label set anymore. This is a problem for the subsequent Cilium TF module because it waits for all CP nodes to be available before the installation.

With your changes:
```bash
$ kgn --show-labels
NAME                  STATUS     ROLES    AGE     VERSION   LABELS
i-0238e1f62a0343992   NotReady   <none>   4m24s   v1.31.4   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=i-0238e1f62a0343992,kubernetes.io/os=linux
i-07316ad6229166609   NotReady   <none>   4m25s   v1.31.4   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=i-07316ad6229166609,kubernetes.io/os=linux
i-0cf8f2aeaa9e65467   NotReady   <none>   4m25s   v1.31.4   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=i-0cf8f2aeaa9e65467,kubernetes.io/os=linux
i-0d9c6cf2b08719873   NotReady   <none>   4m24s   v1.31.4   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=i-0d9c6cf2b08719873,kubernetes.io/os=linux
i-0f6e0c1f46443a8f4   NotReady   <none>   4m26s   v1.31.4   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=i-0f6e0c1f46443a8f4,kubernetes.io/os=linux
```

Before your changes:
```bash
$ kgn --show-labels
NAME              STATUS   ROLES           AGE    VERSION   LABELS
ip-10-0-100-197   Ready    control-plane   102s   v1.31.4   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ip-10-0-100-197,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node.kubernetes.io/exclude-from-external-load-balancers=
ip-10-0-100-199   Ready    <none>          104s   v1.31.4   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ip-10-0-100-199,kubernetes.io/os=linux
ip-10-0-101-164   Ready    control-plane   102s   v1.31.4   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ip-10-0-101-164,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node.kubernetes.io/exclude-from-external-load-balancers=
ip-10-0-101-28    Ready    <none>          104s   v1.31.4   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ip-10-0-101-28,kubernetes.io/os=linux
ip-10-0-102-180   Ready    control-plane   102s   v1.31.4   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ip-10-0-102-180,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node.kubernetes.io/exclude-from-external-load-balancers=
```

PhilipSchmid commented:
@erikvveen I've refactored a few things (see https://github.com/isovalent/terraform-aws-talos/pull/173/commits/3889481359bbbe43666f9bdc95813062e109ff91), and it now works on my end - would you please test it as well and confirm?

I took a few things from:
- https://github.com/siderolabs/contrib/blob/main/examples/terraform/aws/main.tf
- https://www.siderolabs.com/blog/deploying-talos-on-aws-with-cdk/
- https://cloud-provider-aws.sigs.k8s.io/prerequisites/

... and changed the `var.metadata_options` default values back to the AWS TF provider defaults as it seems the AWS Cloud Controller struggles with `http_tokens=required`.

Once you confirm, I'll squash everything, run the CI stuff, and merge it eventually.

erikvveen commented:
@PhilipSchmid running your code as it is resulted for me in nodenames following a mixed pattern of "ip-based naming" and "resource based naming", see:

￼

According AWS (https://cloud-provider-aws.sigs.k8s.io/prerequisites/ ) nodenames should follow either the pattern ip-xxx-xxx-xxx-xxx.ec2.<region>.internal
Or i-1234567890abcdefg or i-1234567890abcdefg.<region>.compute.internal,not a mixture.

Reintroducing the hostname-override to the kubelet :
hostname-override = module.talos_worker_group[each.key].id

makes that the node names follow resource based naming correctly but the “control plane” label "node-role.kubernetes.io/control-plane=" disappears (the tier=control-plane stays). The disappearance of the label prevents aws-cloud-controller-manager from launching. Adding manually the label to the control-planes and adding ProviderID’s to all the nodes, starts the cloud controller manager and initializes it correctly.

Please let me know if you see the same?
Thanks,
Erik

> On Feb 14, 2025, at 3:41 PM, Philip Schmid ***@***.***> wrote:
> 
> 
> PhilipSchmid
>  left a comment 
> (isovalent/terraform-aws-talos#173)
> @erikvveen I've refactored a few things (see 3889481), and it now works on my end - would you please test it as well and confirm?
> 
> I took a few things from:
> 
> https://github.com/siderolabs/contrib/blob/main/examples/terraform/aws/main.tf
> https://www.siderolabs.com/blog/deploying-talos-on-aws-with-cdk/
> https://cloud-provider-aws.sigs.k8s.io/prerequisites/
> ... and changed the var.metadata_options default values back to the AWS TF provider defaults as it seems the AWS Cloud Controller struggles with http_tokens=required.
> 
> Once you confirm, I'll squash everything, run the CI stuff, and merge it eventually.
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.
> 
>  <https://github.com/erikvveen> <https://github.com/isovalent/terraform-aws-talos/commit/3889481359bbbe43666f9bdc95813062e109ff91> <https://github.com/siderolabs/contrib/blob/main/examples/terraform/aws/main.tf>
>  <https://www.siderolabs.com/blog/deploying-talos-on-aws-with-cdk/>
>  <https://cloud-provider-aws.sigs.k8s.io/prerequisites/> <https://github.com/isovalent/terraform-aws-talos/pull/173#issuecomment-2659507695> <https://github.com/notifications/unsubscribe-auth/AJGXPWS4P3CZ45FFMTTUL532PX6BHAVCNFSM6AAAAABURRF4BOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNJZGUYDONRZGU>
> 
> PhilipSchmid
>  left a comment 
> (isovalent/terraform-aws-talos#173)
>  <https://github.com/isovalent/terraform-aws-talos/pull/173#issuecomment-2659507695>
> @erikvveen <https://github.com/erikvveen> I've refactored a few things (see 3889481 <https://github.com/isovalent/terraform-aws-talos/commit/3889481359bbbe43666f9bdc95813062e109ff91>), and it now works on my end - would you please test it as well and confirm?
> 
> I took a few things from:
> 
> https://github.com/siderolabs/contrib/blob/main/examples/terraform/aws/main.tf
> https://www.siderolabs.com/blog/deploying-talos-on-aws-with-cdk/
> https://cloud-provider-aws.sigs.k8s.io/prerequisites/
> ... and changed the var.metadata_options default values back to the AWS TF provider defaults as it seems the AWS Cloud Controller struggles with http_tokens=required.
> 
> Once you confirm, I'll squash everything, run the CI stuff, and merge it eventually.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/isovalent/terraform-aws-talos/pull/173#issuecomment-2659507695>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AJGXPWS4P3CZ45FFMTTUL532PX6BHAVCNFSM6AAAAABURRF4BOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNJZGUYDONRZGU>.
> You are receiving this because you were mentioned.
>

PhilipSchmid commented:
@erikvveen What exactly do you mean by `nodenames following a mixed pattern`? When issuing `kubectl get nodes` all the nodes follow the same IP-based naming pattern:

```bash
$ kubectl get nodes
NAME                                          STATUS   ROLES           AGE   VERSION
ip-10-0-100-145.eu-north-1.compute.internal   Ready    control-plane   86s   v1.31.4
ip-10-0-101-23.eu-north-1.compute.internal    Ready    control-plane   85s   v1.31.4
ip-10-0-101-245.eu-north-1.compute.internal   Ready    <none>          87s   v1.31.4
ip-10-0-102-121.eu-north-1.compute.internal   Ready    <none>          87s   v1.31.4
ip-10-0-102-75.eu-north-1.compute.internal    Ready    control-plane   86s   v1.31.4
```

Yes, it's not directly following the in the docs mentioned `ip-xxx-xxx-xxx-xxx.ec2.<region>.internal` pattern, but it looks like `ip-xxx-xxx-xxx-xxx.<region>.compute.internal` is fine as well:

```bash
$ kubectl get pods -n kube-system -l k8s-app=aws-cloud-controller-manager
NAME                                 READY   STATUS    RESTARTS   AGE
aws-cloud-controller-manager-7pfsh   1/1     Running   0          3m14s
aws-cloud-controller-manager-f2njz   1/1     Running   0          3m14s
aws-cloud-controller-manager-hhpxv   1/1     Running   0          3m14s
```

```bash
kubectl logs -n kube-system aws-cloud-controller-manager-7pfsh                                                    185ms  13:46:10
I0217 12:43:08.447721       1 serving.go:348] Generated self-signed cert in-memory
I0217 12:43:08.744423       1 serving.go:348] Generated self-signed cert in-memory
W0217 12:43:08.744542       1 client_config.go:618] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I0217 12:43:09.310760       1 requestheader_controller.go:244] Loaded a new request header values for RequestHeaderAuthRequestController
I0217 12:43:09.312180       1 aws.go:681] Loading region from metadata service
I0217 12:43:12.469956       1 aws.go:1341] Building AWS cloudprovider
I0217 12:43:12.470022       1 aws.go:681] Loading region from metadata service
I0217 12:43:25.281692       1 tags.go:77] AWS cloud filtering on ClusterID: talos-cute
I0217 12:43:25.281774       1 aws.go:1431] The following IP families will be added to nodes: [ipv4]
I0217 12:43:25.281808       1 controllermanager.go:167] Version: v1.27.1
I0217 12:43:25.291378       1 tlsconfig.go:200] "Loaded serving cert" certName="Generated self signed cert" certDetail="\"localhost@1739796188\" [serving] validServingFor=[127.0.0.1,localhost,localhost] issuer=\"localhost-ca@1739796188\" (2025-02-17 11:43:08 +0000 UTC to 2026-02-17 11:43:08 +0000 UTC (now=2025-02-17 12:43:25.291328146 +0000 UTC))"
I0217 12:43:25.291814       1 configmap_cafile_content.go:202] "Starting controller" name="client-ca::kube-system::extension-apiserver-authentication::client-ca-file"
I0217 12:43:25.292110       1 shared_informer.go:311] Waiting for caches to sync for client-ca::kube-system::extension-apiserver-authentication::client-ca-file
I0217 12:43:25.291835       1 configmap_cafile_content.go:202] "Starting controller" name="client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file"
I0217 12:43:25.292324       1 shared_informer.go:311] Waiting for caches to sync for client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file
I0217 12:43:25.291860       1 requestheader_controller.go:169] Starting RequestHeaderAuthRequestController
I0217 12:43:25.292794       1 shared_informer.go:311] Waiting for caches to sync for RequestHeaderAuthRequestController
I0217 12:43:25.294518       1 named_certificates.go:53] "Loaded SNI cert" index=0 certName="self-signed loopback" certDetail="\"apiserver-loopback-client@1739796189\" [serving] validServingFor=[apiserver-loopback-client] issuer=\"apiserver-loopback-client-ca@1739796189\" (2025-02-17 11:43:08 +0000 UTC to 2026-02-17 11:43:08 +0000 UTC (now=2025-02-17 12:43:25.294479324 +0000 UTC))"
I0217 12:43:25.294592       1 secure_serving.go:210] Serving securely on [::]:10258
I0217 12:43:25.295126       1 leaderelection.go:245] attempting to acquire leader lease kube-system/cloud-controller-manager...
I0217 12:43:25.295666       1 tlsconfig.go:240] "Starting DynamicServingCertificateController"
I0217 12:43:25.392679       1 shared_informer.go:318] Caches are synced for client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file
I0217 12:43:25.392804       1 shared_informer.go:318] Caches are synced for client-ca::kube-system::extension-apiserver-authentication::client-ca-file
I0217 12:43:25.392842       1 shared_informer.go:318] Caches are synced for RequestHeaderAuthRequestController
I0217 12:43:25.393076       1 tlsconfig.go:178] "Loaded client CA" index=0 certName="client-ca::kube-system::extension-apiserver-authentication::client-ca-file,client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file" certDetail="\"\" [serving,client] groups=[kubernetes] issuer=\"<self>\" (2025-02-17 12:39:53 +0000 UTC to 2035-02-15 12:39:53 +0000 UTC (now=2025-02-17 12:43:25.393018897 +0000 UTC))"
I0217 12:43:25.393995       1 tlsconfig.go:200] "Loaded serving cert" certName="Generated self signed cert" certDetail="\"localhost@1739796188\" [serving] validServingFor=[127.0.0.1,localhost,localhost] issuer=\"localhost-ca@1739796188\" (2025-02-17 11:43:08 +0000 UTC to 2026-02-17 11:43:08 +0000 UTC (now=2025-02-17 12:43:25.393964012 +0000 UTC))"
I0217 12:43:25.394906       1 named_certificates.go:53] "Loaded SNI cert" index=0 certName="self-signed loopback" certDetail="\"apiserver-loopback-client@1739796189\" [serving] validServingFor=[apiserver-loopback-client] issuer=\"apiserver-loopback-client-ca@1739796189\" (2025-02-17 11:43:08 +0000 UTC to 2026-02-17 11:43:08 +0000 UTC (now=2025-02-17 12:43:25.394877431 +0000 UTC))"
I0217 12:43:25.395176       1 tlsconfig.go:178] "Loaded client CA" index=0 certName="client-ca::kube-system::extension-apiserver-authentication::client-ca-file,client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file" certDetail="\"\" [serving,client] groups=[kubernetes] issuer=\"<self>\" (2025-02-17 12:39:53 +0000 UTC to 2035-02-15 12:39:53 +0000 UTC (now=2025-02-17 12:43:25.395144595 +0000 UTC))"
I0217 12:43:25.395356       1 tlsconfig.go:178] "Loaded client CA" index=1 certName="client-ca::kube-system::extension-apiserver-authentication::client-ca-file,client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file" certDetail="\"\" [serving,client] issuer=\"<self>\" (2025-02-17 12:39:53 +0000 UTC to 2035-02-15 12:39:53 +0000 UTC (now=2025-02-17 12:43:25.395314115 +0000 UTC))"
I0217 12:43:25.396124       1 tlsconfig.go:200] "Loaded serving cert" certName="Generated self signed cert" certDetail="\"localhost@1739796188\" [serving] validServingFor=[127.0.0.1,localhost,localhost] issuer=\"localhost-ca@1739796188\" (2025-02-17 11:43:08 +0000 UTC to 2026-02-17 11:43:08 +0000 UTC (now=2025-02-17 12:43:25.396098799 +0000 UTC))"
I0217 12:43:25.396915       1 named_certificates.go:53] "Loaded SNI cert" index=0 certName="self-signed loopback" certDetail="\"apiserver-loopback-client@1739796189\" [serving] validServingFor=[apiserver-loopback-client] issuer=\"apiserver-loopback-client-ca@1739796189\" (2025-02-17 11:43:08 +0000 UTC to 2026-02-17 11:43:08 +0000 UTC (now=2025-02-17 12:43:25.396886485 +0000 UTC))"
```

Hence, the [Node name conventions prerequisite](https://cloud-provider-aws.sigs.k8s.io/prerequisites/) seems to be fulfilled. The EC2 instance name on the AWS portal doesn't matter, as far as I can tell.

erikvveen commented:
Ah, I made a stupid mistake. I initialized the cluster still with http_tokens=required
I missed your remark that it struggles with that. 
I corrected it, and it works indeed also with your suggested ip-xxx-xxx-xxx-xxx.<region>.compute.internal pattern (although I must admit I slightly prefer to work in accordance with AWS notation, just to avoid any problems in future or misconceptions).

If you want to squash everything and merge it, it would be great.

Thanks again for your patience with me!

Erik

> On Feb 17, 2025, at 1:48 PM, Philip Schmid ***@***.***> wrote:
> 
> 
> PhilipSchmid
>  left a comment 
> (isovalent/terraform-aws-talos#173)
> @erikvveen What exactly do you mean by nodenames following a mixed pattern? When issuing kubectl get nodes all the nodes follow the same IP-based naming pattern:
> 
> $ kubectl get nodes
> NAME                                          STATUS   ROLES           AGE   VERSION
> ip-10-0-100-145.eu-north-1.compute.internal   Ready    control-plane   86s   v1.31.4
> ip-10-0-101-23.eu-north-1.compute.internal    Ready    control-plane   85s   v1.31.4
> ip-10-0-101-245.eu-north-1.compute.internal   Ready    <none>          87s   v1.31.4
> ip-10-0-102-121.eu-north-1.compute.internal   Ready    <none>          87s   v1.31.4
> ip-10-0-102-75.eu-north-1.compute.internal    Ready    control-plane   86s   v1.31.4
> Yes, it's not directly following the in the docs mentioned ip-xxx-xxx-xxx-xxx.ec2.<region>.internal pattern, but it looks like ip-xxx-xxx-xxx-xxx.<region>.compute.internal is fine as well:
> 
> $ kubectl get pods -n kube-system -l k8s-app=aws-cloud-controller-manager
> NAME                                 READY   STATUS    RESTARTS   AGE
> aws-cloud-controller-manager-7pfsh   1/1     Running   0          3m14s
> aws-cloud-controller-manager-f2njz   1/1     Running   0          3m14s
> aws-cloud-controller-manager-hhpxv   1/1     Running   0          3m14s
> kubectl logs -n kube-system aws-cloud-controller-manager-7pfsh                                                    185ms  13:46:10
> I0217 12:43:08.447721       1 serving.go:348] Generated self-signed cert in-memory
> I0217 12:43:08.744423       1 serving.go:348] Generated self-signed cert in-memory
> W0217 12:43:08.744542       1 client_config.go:618] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
> I0217 12:43:09.310760       1 requestheader_controller.go:244] Loaded a new request header values for RequestHeaderAuthRequestController
> I0217 12:43:09.312180       1 aws.go:681] Loading region from metadata service
> I0217 12:43:12.469956       1 aws.go:1341] Building AWS cloudprovider
> I0217 12:43:12.470022       1 aws.go:681] Loading region from metadata service
> I0217 12:43:25.281692       1 tags.go:77] AWS cloud filtering on ClusterID: talos-cute
> I0217 12:43:25.281774       1 aws.go:1431] The following IP families will be added to nodes: [ipv4]
> I0217 12:43:25.281808       1 controllermanager.go:167] Version: v1.27.1
> I0217 12:43:25.291378       1 tlsconfig.go:200] "Loaded serving cert" certName="Generated self signed cert" ***@***.***\" [serving] validServingFor=[127.0.0.1,localhost,localhost] ***@***.***\" (2025-02-17 11:43:08 +0000 UTC to 2026-02-17 11:43:08 +0000 UTC (now=2025-02-17 12:43:25.291328146 +0000 UTC))"
> I0217 12:43:25.291814       1 configmap_cafile_content.go:202] "Starting controller" name="client-ca::kube-system::extension-apiserver-authentication::client-ca-file"
> I0217 12:43:25.292110       1 shared_informer.go:311] Waiting for caches to sync for client-ca::kube-system::extension-apiserver-authentication::client-ca-file
> I0217 12:43:25.291835       1 configmap_cafile_content.go:202] "Starting controller" name="client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file"
> I0217 12:43:25.292324       1 shared_informer.go:311] Waiting for caches to sync for client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file
> I0217 12:43:25.291860       1 requestheader_controller.go:169] Starting RequestHeaderAuthRequestController
> I0217 12:43:25.292794       1 shared_informer.go:311] Waiting for caches to sync for RequestHeaderAuthRequestController
> I0217 12:43:25.294518       1 named_certificates.go:53] "Loaded SNI cert" index=0 certName="self-signed loopback" ***@***.***\" [serving] validServingFor=[apiserver-loopback-client] ***@***.***\" (2025-02-17 11:43:08 +0000 UTC to 2026-02-17 11:43:08 +0000 UTC (now=2025-02-17 12:43:25.294479324 +0000 UTC))"
> I0217 12:43:25.294592       1 secure_serving.go:210] Serving securely on [::]:10258
> I0217 12:43:25.295126       1 leaderelection.go:245] attempting to acquire leader lease kube-system/cloud-controller-manager...
> I0217 12:43:25.295666       1 tlsconfig.go:240] "Starting DynamicServingCertificateController"
> I0217 12:43:25.392679       1 shared_informer.go:318] Caches are synced for client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file
> I0217 12:43:25.392804       1 shared_informer.go:318] Caches are synced for client-ca::kube-system::extension-apiserver-authentication::client-ca-file
> I0217 12:43:25.392842       1 shared_informer.go:318] Caches are synced for RequestHeaderAuthRequestController
> I0217 12:43:25.393076       1 tlsconfig.go:178] "Loaded client CA" index=0 certName="client-ca::kube-system::extension-apiserver-authentication::client-ca-file,client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file" certDetail="\"\" [serving,client] groups=[kubernetes] issuer=\"<self>\" (2025-02-17 12:39:53 +0000 UTC to 2035-02-15 12:39:53 +0000 UTC (now=2025-02-17 12:43:25.393018897 +0000 UTC))"
> I0217 12:43:25.393995       1 tlsconfig.go:200] "Loaded serving cert" certName="Generated self signed cert" ***@***.***\" [serving] validServingFor=[127.0.0.1,localhost,localhost] ***@***.***\" (2025-02-17 11:43:08 +0000 UTC to 2026-02-17 11:43:08 +0000 UTC (now=2025-02-17 12:43:25.393964012 +0000 UTC))"
> I0217 12:43:25.394906       1 named_certificates.go:53] "Loaded SNI cert" index=0 certName="self-signed loopback" ***@***.***\" [serving] validServingFor=[apiserver-loopback-client] ***@***.***\" (2025-02-17 11:43:08 +0000 UTC to 2026-02-17 11:43:08 +0000 UTC (now=2025-02-17 12:43:25.394877431 +0000 UTC))"
> I0217 12:43:25.395176       1 tlsconfig.go:178] "Loaded client CA" index=0 certName="client-ca::kube-system::extension-apiserver-authentication::client-ca-file,client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file" certDetail="\"\" [serving,client] groups=[kubernetes] issuer=\"<self>\" (2025-02-17 12:39:53 +0000 UTC to 2035-02-15 12:39:53 +0000 UTC (now=2025-02-17 12:43:25.395144595 +0000 UTC))"
> I0217 12:43:25.395356       1 tlsconfig.go:178] "Loaded client CA" index=1 certName="client-ca::kube-system::extension-apiserver-authentication::client-ca-file,client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file" certDetail="\"\" [serving,client] issuer=\"<self>\" (2025-02-17 12:39:53 +0000 UTC to 2035-02-15 12:39:53 +0000 UTC (now=2025-02-17 12:43:25.395314115 +0000 UTC))"
> I0217 12:43:25.396124       1 tlsconfig.go:200] "Loaded serving cert" certName="Generated self signed cert" ***@***.***\" [serving] validServingFor=[127.0.0.1,localhost,localhost] ***@***.***\" (2025-02-17 11:43:08 +0000 UTC to 2026-02-17 11:43:08 +0000 UTC (now=2025-02-17 12:43:25.396098799 +0000 UTC))"
> I0217 12:43:25.396915       1 named_certificates.go:53] "Loaded SNI cert" index=0 certName="self-signed loopback" ***@***.***\" [serving] validServingFor=[apiserver-loopback-client] ***@***.***\" (2025-02-17 11:43:08 +0000 UTC to 2026-02-17 11:43:08 +0000 UTC (now=2025-02-17 12:43:25.396886485 +0000 UTC))"
> Hence, the Node name conventions prerequisite seems to be fulfilled. The EC2 instance name on the AWS portal doesn't matter, as far as I can tell.
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.
> 
>  <https://github.com/erikvveen> <https://cloud-provider-aws.sigs.k8s.io/prerequisites/> <https://github.com/isovalent/terraform-aws-talos/pull/173#issuecomment-2663024624> <https://github.com/notifications/unsubscribe-auth/AJGXPWWVGVFCI2GMVUBAVCT2QHLDBAVCNFSM6AAAAABURRF4BOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNRTGAZDINRSGQ>
> 
> PhilipSchmid
>  left a comment 
> (isovalent/terraform-aws-talos#173)
>  <https://github.com/isovalent/terraform-aws-talos/pull/173#issuecomment-2663024624>
> @erikvveen <https://github.com/erikvveen> What exactly do you mean by nodenames following a mixed pattern? When issuing kubectl get nodes all the nodes follow the same IP-based naming pattern:
> 
> $ kubectl get nodes
> NAME                                          STATUS   ROLES           AGE   VERSION
> ip-10-0-100-145.eu-north-1.compute.internal   Ready    control-plane   86s   v1.31.4
> ip-10-0-101-23.eu-north-1.compute.internal    Ready    control-plane   85s   v1.31.4
> ip-10-0-101-245.eu-north-1.compute.internal   Ready    <none>          87s   v1.31.4
> ip-10-0-102-121.eu-north-1.compute.internal   Ready    <none>          87s   v1.31.4
> ip-10-0-102-75.eu-north-1.compute.internal    Ready    control-plane   86s   v1.31.4
> Yes, it's not directly following the in the docs mentioned ip-xxx-xxx-xxx-xxx.ec2.<region>.internal pattern, but it looks like ip-xxx-xxx-xxx-xxx.<region>.compute.internal is fine as well:
> 
> $ kubectl get pods -n kube-system -l k8s-app=aws-cloud-controller-manager
> NAME                                 READY   STATUS    RESTARTS   AGE
> aws-cloud-controller-manager-7pfsh   1/1     Running   0          3m14s
> aws-cloud-controller-manager-f2njz   1/1     Running   0          3m14s
> aws-cloud-controller-manager-hhpxv   1/1     Running   0          3m14s
> kubectl logs -n kube-system aws-cloud-controller-manager-7pfsh                                                    185ms  13:46:10
> I0217 12:43:08.447721       1 serving.go:348] Generated self-signed cert in-memory
> I0217 12:43:08.744423       1 serving.go:348] Generated self-signed cert in-memory
> W0217 12:43:08.744542       1 client_config.go:618] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
> I0217 12:43:09.310760       1 requestheader_controller.go:244] Loaded a new request header values for RequestHeaderAuthRequestController
> I0217 12:43:09.312180       1 aws.go:681] Loading region from metadata service
> I0217 12:43:12.469956       1 aws.go:1341] Building AWS cloudprovider
> I0217 12:43:12.470022       1 aws.go:681] Loading region from metadata service
> I0217 12:43:25.281692       1 tags.go:77] AWS cloud filtering on ClusterID: talos-cute
> I0217 12:43:25.281774       1 aws.go:1431] The following IP families will be added to nodes: [ipv4]
> I0217 12:43:25.281808       1 controllermanager.go:167] Version: v1.27.1
> I0217 12:43:25.291378       1 tlsconfig.go:200] "Loaded serving cert" certName="Generated self signed cert" ***@***.***\" [serving] validServingFor=[127.0.0.1,localhost,localhost] ***@***.***\" (2025-02-17 11:43:08 +0000 UTC to 2026-02-17 11:43:08 +0000 UTC (now=2025-02-17 12:43:25.291328146 +0000 UTC))"
> I0217 12:43:25.291814       1 configmap_cafile_content.go:202] "Starting controller" name="client-ca::kube-system::extension-apiserver-authentication::client-ca-file"
> I0217 12:43:25.292110       1 shared_informer.go:311] Waiting for caches to sync for client-ca::kube-system::extension-apiserver-authentication::client-ca-file
> I0217 12:43:25.291835       1 configmap_cafile_content.go:202] "Starting controller" name="client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file"
> I0217 12:43:25.292324       1 shared_informer.go:311] Waiting for caches to sync for client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file
> I0217 12:43:25.291860       1 requestheader_controller.go:169] Starting RequestHeaderAuthRequestController
> I0217 12:43:25.292794       1 shared_informer.go:311] Waiting for caches to sync for RequestHeaderAuthRequestController
> I0217 12:43:25.294518       1 named_certificates.go:53] "Loaded SNI cert" index=0 certName="self-signed loopback" ***@***.***\" [serving] validServingFor=[apiserver-loopback-client] ***@***.***\" (2025-02-17 11:43:08 +0000 UTC to 2026-02-17 11:43:08 +0000 UTC (now=2025-02-17 12:43:25.294479324 +0000 UTC))"
> I0217 12:43:25.294592       1 secure_serving.go:210] Serving securely on [::]:10258
> I0217 12:43:25.295126       1 leaderelection.go:245] attempting to acquire leader lease kube-system/cloud-controller-manager...
> I0217 12:43:25.295666       1 tlsconfig.go:240] "Starting DynamicServingCertificateController"
> I0217 12:43:25.392679       1 shared_informer.go:318] Caches are synced for client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file
> I0217 12:43:25.392804       1 shared_informer.go:318] Caches are synced for client-ca::kube-system::extension-apiserver-authentication::client-ca-file
> I0217 12:43:25.392842       1 shared_informer.go:318] Caches are synced for RequestHeaderAuthRequestController
> I0217 12:43:25.393076       1 tlsconfig.go:178] "Loaded client CA" index=0 certName="client-ca::kube-system::extension-apiserver-authentication::client-ca-file,client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file" certDetail="\"\" [serving,client] groups=[kubernetes] issuer=\"<self>\" (2025-02-17 12:39:53 +0000 UTC to 2035-02-15 12:39:53 +0000 UTC (now=2025-02-17 12:43:25.393018897 +0000 UTC))"
> I0217 12:43:25.393995       1 tlsconfig.go:200] "Loaded serving cert" certName="Generated self signed cert" ***@***.***\" [serving] validServingFor=[127.0.0.1,localhost,localhost] ***@***.***\" (2025-02-17 11:43:08 +0000 UTC to 2026-02-17 11:43:08 +0000 UTC (now=2025-02-17 12:43:25.393964012 +0000 UTC))"
> I0217 12:43:25.394906       1 named_certificates.go:53] "Loaded SNI cert" index=0 certName="self-signed loopback" ***@***.***\" [serving] validServingFor=[apiserver-loopback-client] ***@***.***\" (2025-02-17 11:43:08 +0000 UTC to 2026-02-17 11:43:08 +0000 UTC (now=2025-02-17 12:43:25.394877431 +0000 UTC))"
> I0217 12:43:25.395176       1 tlsconfig.go:178] "Loaded client CA" index=0 certName="client-ca::kube-system::extension-apiserver-authentication::client-ca-file,client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file" certDetail="\"\" [serving,client] groups=[kubernetes] issuer=\"<self>\" (2025-02-17 12:39:53 +0000 UTC to 2035-02-15 12:39:53 +0000 UTC (now=2025-02-17 12:43:25.395144595 +0000 UTC))"
> I0217 12:43:25.395356       1 tlsconfig.go:178] "Loaded client CA" index=1 certName="client-ca::kube-system::extension-apiserver-authentication::client-ca-file,client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file" certDetail="\"\" [serving,client] issuer=\"<self>\" (2025-02-17 12:39:53 +0000 UTC to 2035-02-15 12:39:53 +0000 UTC (now=2025-02-17 12:43:25.395314115 +0000 UTC))"
> I0217 12:43:25.396124       1 tlsconfig.go:200] "Loaded serving cert" certName="Generated self signed cert" ***@***.***\" [serving] validServingFor=[127.0.0.1,localhost,localhost] ***@***.***\" (2025-02-17 11:43:08 +0000 UTC to 2026-02-17 11:43:08 +0000 UTC (now=2025-02-17 12:43:25.396098799 +0000 UTC))"
> I0217 12:43:25.396915       1 named_certificates.go:53] "Loaded SNI cert" index=0 certName="self-signed loopback" ***@***.***\" [serving] validServingFor=[apiserver-loopback-client] ***@***.***\" (2025-02-17 11:43:08 +0000 UTC to 2026-02-17 11:43:08 +0000 UTC (now=2025-02-17 12:43:25.396886485 +0000 UTC))"
> Hence, the Node name conventions prerequisite <https://cloud-provider-aws.sigs.k8s.io/prerequisites/> seems to be fulfilled. The EC2 instance name on the AWS portal doesn't matter, as far as I can tell.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/isovalent/terraform-aws-talos/pull/173#issuecomment-2663024624>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AJGXPWWVGVFCI2GMVUBAVCT2QHLDBAVCNFSM6AAAAABURRF4BOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNRTGAZDINRSGQ>.
> You are receiving this because you were mentioned.
>
